lightning/src/chain/chainmonitor.rs

```diff
@@ -387,7 +598,12 @@ where C::Target: chain::Filter,
 				monitor.load_outputs_to_watch(chain_source);
 			}
 		}
-		entry.insert(MonitorHolder { monitor });
+		entry.insert(MonitorHolder {
+			monitor,
+			pending_monitor_updates: Mutex::new(pending_monitor_updates),
+			channel_perm_failed: AtomicBool::new(false),
```

Comment:
I don't think we care? If we get `locked_from_offchain` that means ChannelManager told us the channel is closed, so we're probably not gonna get into `watch_channel` at that point. The only case where it can be perm-failed at this point is if we failed to persist above, but then we'd have returned early already.